tests/fast-integration/eventing-test/eventing-test.js

```diff
@@ -92,20 +112,36 @@ describe('Eventing tests', function() {
 
   // Tests
   context('with Nats backend', function() {
+    it('Switch Eventing Backend to Nats', async function() {
+      const currentBackend = await getEventingBackend();
+      if (currentBackend && currentBackend.toLowerCase() === natsBackend) {
+        return;
+      }
+      await switchEventingBackend(backendK8sSecretName, backendK8sSecretNamespace, natsBackend);
+    });
     it('Wait until subscriptions are ready', async function() {
       await waitForSubscriptionsTillReady(testNamespace);
     });
     // Running Eventing end-to-end tests
-    eventingE2ETestSuite();
+    eventingE2ETestSuite(natsBackend);
     // Running Eventing tracing tests
     eventingTracingTestSuite();
     // Running Eventing Monitoring tests
-    eventingMonitoringTest('nats');
+    eventingMonitoringTest(natsBackend);
   });
 
   context('with BEB backend', function() {
+    // skip publishing cloud events for beb backend when event mesh credentials file is missing
+    if (eventMeshNamespace === undefined) {
```

Comment:
In which cases can `eventMeshNamespace` be undefined ?

Reply:
It's exactly then undefined, when it's written in the comment right above :) When I cannot fetch the eventMeshNamespace, I return an undefined which is considered as a sign of empty namespace.

Reply:
I quickly checked the documentation of mockajs and found a [way to skip a test at runtime](https://mochajs.org/#inclusive-tests) using the test framework itself. Just returning and not executing the `It` block works of course, but will not show as a test (with pending status) in the test report. 

```js
it('should only test in the correct environment', function() {
  if (/* check test environment */) {
    // make assertions
  } else {
    this.skip();
  }
});
```

For more infos see this fiddle: https://codepen.io/alexpyzhianov/pen/KVbeyO and add `this.skip()` to  `it('is evergreen'`

<img width="180" alt="Screenshot 2022-02-21 at 6 46 01 PM" src="https://user-images.githubusercontent.com/8013145/155005004-e8e26e3c-dad1-436f-81db-9e5dbd43edfe.png">

I personally prefer the version where the test framework skips the execution of a test because it is more visible to the reader of the test report log that a certain test was skipped.

Reply:
I added them where possible, thanks for the tip